pkg/option/config.go

```diff
@@ -431,6 +431,9 @@ const (
 	// ProxySocketDir is the directory path to store proxy sockets
 	ProxySocketDir = "proxy-socket-dir"
 
+	// EmbeddedProxyEnabled defines whether the embedded Cilium Proxy should be enabled or not.
+	EmbeddedProxyEnabled = "proxy-embedded-enabled"
```

Comment:
This also relates to a larger uncertainty I have on the naming. IMO it would be nice to have the container named as `cilium-envoy` to give an outward acknowledgement that we are using Envoy, and that we are not trying to hide it behind `cilium-proxy`. I know this deviates from the github repo naming, but so does `cilium-agent` from `cilium/cilium`.

Reply:
tend to use the proposal of @sayboras and go with `external-envoy-proxy` - with the additional benefit of naming it after the mode which eventually will survive (even after embedded mode has been deleted)

Reply:
:100: